lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go

```diff
@@ -99,16 +100,30 @@ func listSubscribersHandler(c echo.Context) error {
 	}
 
 	// First check for query params to filter by
+	var imsis []string
 	if msisdn := c.QueryParam(ParamMSISDN); msisdn != "" {
-		imsi, err := subscriberdb.GetIMSIForMSISDN(networkID, msisdn)
+		queryIMSI, err := subscriberdb.GetIMSIForMSISDN(networkID, msisdn)
 		if err != nil {
 			return makeErr(err)
 		}
-		sub, err := loadSubscribers(networkID, imsi)
+		imsis = []string{queryIMSI}
+	}
+	if ip := c.QueryParam(ParamIP); ip != "" {
+		queryIMSIs, err := subscriberdb.GetIMSIsForIP(networkID, ip)
+		if err != nil {
+			return makeErr(err)
+		}
+		if len(queryIMSIs) == 0 {
+			return makeErr(merrors.ErrNotFound)
+		}
+		imsis = queryIMSIs
+	}
+	if len(imsis) != 0 {
```

Comment:
I think you need to handle the edge case where you return an empty response for one of these filtered queries (i.e. imsis is empty but you actually want to return an empty json map)

Reply:
I was doing a 404 for that but you're right empty map makes more sense